crates/net/eth-wire-types/src/message.rs

```diff
@@ -252,6 +259,14 @@ impl<N: NetworkPrimitives> EthMessage<N> {
             Self::NodeData(_) => EthMessageID::NodeData,
             Self::GetReceipts(_) => EthMessageID::GetReceipts,
             Self::Receipts(_) => EthMessageID::Receipts,
+            Self::Other(bytes) => {
+                // If the message is unknown, assume the first byte is the message ID
+                if bytes.is_empty() {
+                    EthMessageID::Other(0xFF) // Default unknown ID
+                } else {
+                    EthMessageID::Other(bytes[0]) // Extract ID from first byte
+                }
```

Comment:
this is actually a bit more problematic, 
need to think about how to fix this, but for now this is ok.